getting-started/gopher-run/cmd/main.go

```diff
@@ -45,15 +56,94 @@ func main() {
 	}
 	http.HandleFunc("/leaderboard/post", a.addScore)
 	http.HandleFunc("/leaderboard/get", a.topScores)
-	http.Handle("/", http.StripPrefix("/", http.FileServer(http.Dir("static"))))
+	http.HandleFunc("/predict", a.predictionRequest)
+	http.HandleFunc("/pldata", a.addPlayData)
+	http.HandleFunc("/bggenerator", a.sendGeneratedBackground)
+	http.Handle("/", http.StripPrefix("/", http.FileServer(http.Dir("static/gorun/"))))
+	// a.submitTrainingJob(context.Background())
 	port := os.Getenv("PORT")
 	if port == "" {
 		port = "8080"
 	}
-	fmt.Printf("Starting server: localhost:%v\n", port)
+	fmt.Printf("Starting server on port %v\n", port)
 	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%s", port), nil))
 }
 
+func (a *app) predictionRequest(w http.ResponseWriter, r *http.Request) {
+	client, err := google.DefaultClient(oauth2.NoContext,
+		"https://www.googleapis.com/auth/cloud-platform")
+	if err != nil {
+		log.Printf("DefaultClient: %v", err)
+	}
+	resp, err := client.Post("https://ml.googleapis.com/v1/projects/maralder-start/models/playerdata_linear_classification:predict", "application/json", r.Body)
+	if err != nil {
+		log.Printf("client.Post: %v", err)
+	}
+	io.Copy(w, resp.Body)
+	resp.Body.Close()
+}
+
+// Concatenate data from the top runs and start a Cloud ML training job on it
+func (a *app) submitTrainingJob(ctx context.Context) error {
+	bkt := a.bucket
+	topPlayers, err := leaderboard.TopScores(ctx, a.fsClient)
+	if err != nil {
+		log.Printf("leaderboard.TopScores: %v", err)
+		return err
+	}
+	data := []byte(`idle,0,0,0,0,0,0,0,0,0,0,0
```

Comment:
Add a comment saying why this is being done.

Reply:
Done